include/boost/gil/image_processing/morphology.hpp

```diff
@@ -0,0 +1,285 @@
+//
+// Copyright 2021 Prathamesh Tagore <prathameshtagore@gmail.com>
+//
+// Use, modification and distribution are subject to the Boost Software License,
+// Version 1.0. (See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt)
+//
+
+#ifndef BOOST_GIL_IMAGE_PROCESSING_MORPHOLOGY_HPP
+#define BOOST_GIL_IMAGE_PROCESSING_MORPHOLOGY_HPP
+#include <boost/gil/extension/numeric/kernel.hpp>
+#include <boost/gil/gray.hpp>
+#include <boost/gil/image_processing/threshold.hpp>
+
+namespace gil = boost::gil;
+namespace boost{
+    namespace gil{
+        enum class morphological_operations
+            {
+                dilation,
+                erosion,
+            };
+namespace detail{
+/// \addtogroup ImageProcessing
+/// @{
+
+/// \brief Implements morphological operations at pixel level.This function compares neighbouring 
+/// pixel values according to the kernel and choose minimum/mamximum neighbouring pixel value and 
+/// assigns it to the pixel under consideration.
+/// \param src_view - Source/Input image view.
+/// \param dst_view - View which stores the final result of operations performed by this function.
+/// \param kernel - Kernel matrix/structuring element containing 0's and 1's which will be used for applying the required morphological operation.
+/// \param identifier - Indicates the type of morphological operation to be applied.
+/// \tparam SrcView type of source image.
+/// \tparam DstView type of output image.
+/// \tparam Kernel type of structuring element.
+template <typename SrcView, typename DstView, typename Kernel>
+void morph_impl(SrcView const& src_view, DstView const& dst_view, Kernel const& kernel,
+                morphological_operations identifier)
+{
+    std::ptrdiff_t flip_ker_row, flip_ker_col, row_boundary, col_boundary;
+    typename channel_type<typename SrcView::value_type>::type target_element;
```

Comment:
If `target_element` is used inside the nested look, then narrow its scope to the loop

Reply:
```target_element``` is used for comparing all values of image view pixels which are overlapped with the structuring element in one single operation . Therefore, it must remain same during all such comparisons between overlapped values, this stops me from narrowing its scope .

Reply:
Right, I missed that.